### PR TITLE
FIX: Fixed wrong `Display Index` value for touchscreen events (ISXB-1101)

### PR DESCRIPTION
**Pull request by bmalrat**
### Description

When reading the `Display Index` of touch event is add 2048 to the value.
https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-1101
It uses the inherited pointer's `ushort` definition but for touchscreen it's a `byte`.

### Changes made

Fixed the mapping of the `Display Index` field that doesn't match for touch event.

### Testing

local testing on multiple screen

### Risk

_Please describe the potential risks of your changes for the reviewers._

### Checklist

Before review:

- [X] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs [hunk 1]
```diff
@@ -374,6 +374,7 @@ internal unsafe struct TouchscreenState : IInputStateTypeInfo
         [InputControl(name = "pressure", useStateFrom = "primaryTouch/pressure")]
         [InputControl(name = "radius", useStateFrom = "primaryTouch/radius")]
         [InputControl(name = "press", useStateFrom = "primaryTouch/phase", layout = "TouchPress", synthetic = true, usages = new string[0])]
+        [InputControl(name = "displayIndex", useStateFrom = "primaryTouch/displayIndex", format = "BYTE")] // added format to override the Pointer's USHT value
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs:377]: Looks like a good fix. Thanks for adding the comment to say why format is added to make sure its not removed in future refactorings.

File: Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs [hunk 2]
```diff
@@ -559,7 +560,6 @@ protected override void FinishSetup()
             base.FinishSetup();
 
             primaryTouch = GetChildControl<TouchControl>("primaryTouch");
-            displayIndex = primaryTouch.displayIndex;
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs:562]: Why is this line removed ?

I'd suggest Hakan review this PR since he added this code:
https://github.com/Unity-Technologies/InputSystem/pull/1623

Comment (bmalrat) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Touchscreen.cs:562]: It's not needed any more since I added the other attributes which override display index and use the state from the primary touch.
Like other overridden values like pressure Wich are already mapped in the base Pointer class.

**Review by stefanunity — APPROVED** [review]
-works with two displays (1 a touch) showing correct displayIndex
-touch works as per expectations from the user now

**Review by lyndon-unity — APPROVED** [review]

**Review by ekcoh — APPROVED** [review]
LGTM, thanks for fixing this
